streamlit/streamlit_app.py

```diff
@@ -0,0 +1,289 @@
+"""An example of Streamlit leveraging Wise pizza."""
+
+import streamlit as st
+import os, sys
+import pandas as pd
+
+import warnings
+warnings.filterwarnings("ignore")
+
+root_path = os.path.realpath('../..')
+
+# this assumes that all of the following files are checked in the same directory
+sys.path.append(os.path.join(root_path,"wise-pizza"))
+
+# create data-related directories
+data_dir = os.path.realpath(os.path.join(root_path, 'wise-pizza/data'))
+if not os.path.isdir(data_dir):
+    os.mkdir(data_dir)
+
+from wise_pizza import explain_levels, explain_changes_in_totals, explain_changes_in_average
+
+# False if you want nice interactive plots
+# True if you want static plots (Doesn't work on all platforms yet)
+plot_is_static = False
+
+
+# SETTING PAGE CONFIG TO WIDE MODE AND ADDING A TITLE AND FAVICON
+st.set_page_config(layout="wide", page_title="Wise Pizza", page_icon=":pizza:")
+
+st.title('Wise Pizza powered by Streamlit')
+st.text('Only categorical columns are accepted, bucket the numeric ones if you wanna use those')
+
+# upload the file from the computer
+def load_data_upload():
+    uploaded_file = st.file_uploader("Choose a file")
+    if not uploaded_file:
+        st.warning('Please input a dataset.')
+        st.stop()
+    st.success('Dataset inputted.')
+    data = pd.read_csv(uploaded_file)
+    return data
+
+on = st.toggle('Use sample data from Github')
+url_data = (r'https://raw.githubusercontent.com/transferwise/wise-pizza/main/data/synth_data.csv')
+
+# select the datasource, either local or from github
+if on:
+    st.write(f'Downloading data from {url_data}!')
+    df = pd.read_csv(url_data)
+else:
+    df=load_data_upload()
+
+# show dataset preview
+st.text('Table preview')
+st.table(df.head(10))
+
+# ask the user via streamlit for the target column
+totals = st.selectbox(
+   "What is the target column that you want to analyse? e.g. GMV/revenue",
+   df.columns,
+   index=None,
+   placeholder="Select target column",
+)
+st.write('You selected:', totals)
+
+# ask the user via streamlit for the size column
+size = st.selectbox(
+   "What is the volume column of your dataset? e.g. number of users/transactions",
```

Comment:
It would make sense to keep the descriptions in streamlit a bit closer to how it is done in the jupyter notebook:
total_name: Name of column that contains totals per segment
size_name: Name of column containing segment size


 In particular, 'What is the volume column...?' may be hard to understand